### PR TITLE
trim lightning dev version in stable deps

### PR DESCRIPTION
**Pull request by timmysilv**
**Context:**
Every time there's a new merge to lightning master, the version on the test PyPI repo gets bumped. This causes noisy commits like [this one](https://github.com/PennyLaneAI/pennylane/pull/5250/commits/0a537929f8224b5fdc8d5717324e8897270a4fd9) for test suites that use the latest master version of lightning. The versions look something like `"PennyLane_Lightning==0.35.0.dev16"`, so we'd see a bump from `dev16` to `dev17`. See [this commit](https://github.com/PennyLaneAI/pennylane/pull/5250/commits/329670ee28407e273a941523f724e9ae23a4c8d7) as an example of this change in action.

**Description of change:**
Always trim lightning dev versions to the stable version

**Benefits:**
- We don't get notified whenever there's a new merge to lightning master
- The file can be pip installed (the dev versions are only on the test pypi repo, and we usually wouldn't access them so the version wouldn't resolve)

**Possible drawbacks:**
- We might want to know exactly which dev version of lightning is installed, but if it's due to a recent issue, we could probably figure it out anyway...
- `pip install` will fail until that major version of lightning is actually released

[sc-57574]

### CODE REVIEW

File: .github/workflows/install_deps/action.yml [hunk 1]
```diff
@@ -103,7 +103,7 @@ runs:
     - name: Freeze dependencies
       shell: bash
       if: inputs.requirements_file != ''
-      run: pip freeze | grep -v "file:///" > ${{ inputs.requirements_file }}
+      run: pip freeze | grep -v "file:///" | sed 's/\(Lightning==[0-9\.]\+\)\.dev[0-9]\+/\1/' > ${{ inputs.requirements_file }}
```
Comment (Mandrenkov) [inline .github/workflows/install_deps/action.yml:106]: # Suggestion
We can simplify this a bit:
```suggestion
      run: pip freeze | grep -v "file:///" | sed 's/\(Lightning==[0-9\.]\+\)\.dev/\1/' > ${{ inputs.requirements_file }}
```

**Review by Mandrenkov — APPROVED** [review]
Overall, looks good to me! 🚀 

For bonus points, I would consider adding:
* An `i` flag to ensure the match is case-insensitive.
* A `g` flag to ensure multiple instances can be matched.
* The full package name to avoid collisions (however unlikely they may be).

Note that we can also ignore anything after `dev` since it is not important:
```
's/^\(pennylane[-_]lightning==[0-9\.]\+\)\.dev/\1/ig'
```

**Comment by Mandrenkov** [comment]
It would be nice to test out any changes to the command before merging!

**Comment by timmysilv** [comment]
I have updated it accordingly.
- aded the `g` and `I` flags for global and case-insensitive matching
- included the full package name
- had to keep the trailing digits otherwise they'd be added afterwards

**Comment by timmysilv** [comment]
[here](https://github.com/PennyLaneAI/pennylane/pull/5250/commits/329670ee28407e273a941523f724e9ae23a4c8d7)'s a commit triggered by this branch to validate that it works

**Review by Mandrenkov — APPROVED** [review]
Perfect, looks great! 🥳

**Review by Alex-Preciado — APPROVED** [review]
Looks good 🙂

**Pull request merged**